_posts/2024-12-30-debezium-oracledb-logminer.md

```diff
@@ -0,0 +1,313 @@
+  
+  GRANT CREATE TABLE TO c##dbzuser CONTAINER=ALL;
+  GRANT LOCK ANY TABLE TO c##dbzuser CONTAINER=ALL;
+  GRANT CREATE SEQUENCE TO c##dbzuser CONTAINER=ALL;
+  
+  GRANT EXECUTE ON DBMS_LOGMNR TO c##dbzuser CONTAINER=ALL;
+  GRANT EXECUTE ON DBMS_LOGMNR_D TO c##dbzuser CONTAINER=ALL;
+  
+  GRANT SELECT ON V_$LOG TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$LOG_HISTORY TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$LOGMNR_LOGS TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$LOGMNR_CONTENTS TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$LOGMNR_PARAMETERS TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$LOGFILE TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$ARCHIVED_LOG TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$ARCHIVE_DEST_STATUS TO c##dbzuser CONTAINER=ALL;
+  GRANT SELECT ON V_$TRANSACTION TO c##dbzuser CONTAINER=ALL;
+  
+  EXIT;
+
+  '''
+
+# Create Initial Test Data
+
+  Connect to Oracle DB using command:
+
+  '''
+  $> sqlplus ‘/ as sysdba’
+  SQL> connect c##dbzuser/dbz
+  SQL> alter session set container=ORCLPDB1;
+  '''
+
+  Create table and some initial sample data.
+
+  '''
+  CREATE TABLE customers (id number(9,0) primary key, name varchar2(50)); INSERT INTO customers VALUES (1001, 'Jane Doe'); 
+  INSERT INTO customers VALUES (1002, 'Bob Willy'); 
+  INSERT INTO customers VALUES (1003, 'Eddie Murphy'); 
+  INSERT INTO customers VALUES (1004, 'Anne Mary'); 
+  COMMIT;
+
+  '''
+
+  Set the table’s supplemental log level:
+
+  '''
+  ALTER TABLE customers ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS;
+
+  '''
+
+# Deploy Debezium Oracle Connector
+
+  Create a source connector file “source_logMiner.json” with configuration:
+
+  '''
+  {
+  "name": "debezium-log-miner",
+  "config": {
+    "connector.class": "io.debezium.connector.oracle.OracleConnector",
+    "tasks.max": "1",
+    "topic.prefix" : "server1",
+    "database.hostname": "oracle-db",
+    "database.port": "1521",
+    "database.user": "c##dbzuser",
+    "database.password": "dbz",
+    "database.dbname": "ORCLCDB",
+    "database.pdb.name": "ORCLPDB1",
+    "database.server.name": "server1",
+    "table.include.list": "C##DBZUSER.CUSTOMERS",
+    "database.history.kafka.bootstrap.servers": "broker:29092",
+    "database.history.kafka.topic": "schema-changes",
+    "schema.history.internal.kafka.bootstrap.servers" : "broker:29092",
+    "schema.history.internal.kafka.topic": "schema-changes.new_table"
+  }
+}
+'''
+
+Save above configuration and deploy the source connector 
+
+'''
+$> curl -i -X POST -H "Accept:application/json" \ -H "Content-Type:application/json" \ localhost:8083/connectors \ -d @r source_logMiner.json | jq
+'''
+
+Once the source connector registration is successful, open Confluent Platform Control Center using http://<host-ip>:9021 in a browser. Navigate to Overview → Topics. You should see a topic “server1.C__DBZUSER.CUSTOMERS’ listed there. Click on the topic and verify the contents of topic.
```

Comment:
What do we expect to see? Also, does the user needs to perform specific actions to see the data coming from the DB?